File: robocode2-server/src/main/java/net/robocode2/events/BulletHitBotEvent.java
package net.robocode2.events;

import lombok.Value;
import net.robocode2.model.Bullet;

/**
 * Event sent when a bullet hits a bot.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BulletHitBotEvent implements Event {

	/** Turn number when event occurred */
	int turnNumber;

	/** Bullet that hit the bot */
	Bullet bullet;

	/** Bot id of the victim that was hit by the bullet */
	int victimId;

	/** Damage dealt to the victim */
	double damage;

	/** New energy level of the victim after damage */
	double energy;
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/event/BulletHitBotEvent.kt
package dev.robocode.tankroyale.server.event

import dev.robocode.tankroyale.server.model.BotId
import dev.robocode.tankroyale.server.model.IBullet

/** Event sent when a bullet hits a bot. */
class BulletHitBotEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bullet that hit the bot */
    val bullet: IBullet,

    /** Bot id of the victim that was hit by the bullet */
    val victimId: BotId,

    /** Damage dealt to the victim */
    val damage: Double,

    /** New energy level of the victim after damage */
    val energy: Double,
) : Event()